src/Middleware/SpaServices.Extensions/src/DevelopmentServer/DevelopmentServerMiddlewareExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Builder;
+using System;
+
+namespace Microsoft.AspNetCore.SpaServices.DevelopmentServer
+{
+    /// <summary>
+    /// Extension methods for enabling React development server middleware support.
+    /// </summary>
+    public static class DevelopmentServerMiddlewareExtensions
+    {
+        /// <summary>
+        /// Handles requests by passing them through to an instance of a development npm web server.
+        /// This means you can always serve up-to-date CLI-built resources without having
+        /// to run the npm web server manually.
+        ///
+        /// This feature should only be used in development. For production deployments, be
+        /// sure not to enable the npm web server.
+        /// </summary>
+        /// <param name="spaBuilder">The <see cref="ISpaBuilder"/>.</param>
+        /// <param name="npmScript">The name of the script in your package.json file that launches the web server.</param>
+        /// <param name="waitText">The text snippet identified during the build to indicate the Development Server has compiled and is ready.</param>
+        /// <param name="serverName">The name of the Server used in the Console.</param>
+        public static void UseDevelopmentServer(
+            this ISpaBuilder spaBuilder,
+            string npmScript,
+            string waitText,
+            string serverName = "App")
```

Comment:
To me, these APIs feel extremely over-specialized. What if the "ready" signal isn't some static text you can look for? What if the way to start it isn't running an NPM script?

What I'd expect instead would be to decompose this into lower-level building blocks, e.g.:

 * Some method for starting an NPM script and getting back the `Process`
 * Some method for transforming the standard output stream from a `Process` into an async-enumerable-of-string
 * `UseReactDevelopmentServer` and `UseAngularCli` methods that call the "start NPM task" method, then awaits until it outputs the expected "I'm running now" string. These can each contain their own instructions to output text like "Starting Angular CLI server...", so no need to make these into parameters on something else.